pallets/pallet-dyn-filter/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2022 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+//! # Dynamic Call Filter
+//!
+//! Enable or disable specific features without a runtime upgrade.
+//!
+//! - [`Pallet`]
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod default_weights;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+pub mod setting;
+
+pub use crate::{default_weights::WeightInfo, pallet::*};
+
+#[frame_support::pallet]
+pub mod pallet {
+
+	use frame_support::{
+		pallet_prelude::*,
+		traits::{Contains, StorageVersion},
+	};
+	use frame_system::pallet_prelude::*;
+	use kilt_support::traits::EnabledFunctionality;
+
+	use crate::{setting::FilterSettings, WeightInfo};
+
+	pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		/// The origin check for all DID calls inside this pallet.
+		type EnsureOrigin: EnsureOrigin<<Self as frame_system::Config>::Origin>;
```

Comment:
W: Give this a better naming like
```suggestion
		type ApproveOrigin: EnsureOrigin<<Self as frame_system::Config>::Origin>;
```